### ISSUE TITLE
Bug: [no-unnecessary-condition] Wrongly reports "always truthy" on class instance property

### ISSUE THREAD

**Issue by MetabaronCZ** [typescript-eslint/typescript-eslint#12423]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [x] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=6.0.3&fileType=.ts&code=KYDwDg9gTgLgBAYwDYEMDOa4HEUFthwDeAsAFBxxhQCWAbijAdWgEoCuAdh9RwOYBccAEYQISYCg4BuMmQoIIHNDChsEMaAAoAlETkU4MABbMAdM3ZcevOAF44AMxRI0wGeTgBfWR7BshSNQIcMoosDqCtBDUACZ6HhTUDnCaxmYWnNx8uiQJBsZQEADucBzAJQCiUIVQmgDkAMIQbEhxHBDwobDYeMCCAIJIUBIxAJ5wqlZ8AIR12u4GXvoUaWjmrJnWdoaqbj6L4vAw1PjbAAzuy3BFJuIpq%2BuWWbw5VxQ31HepJwQAfPZnV55RYPNhgGIMYA6BaLfI-ADU8MuwIMSRS0weGSmL3isNhQmGKAA1jC8d4UeS8Q9hhwYsBavMrpSlqR9H4AkE4GCIYwInAorFcbDMRtsdsALIMIymKCSGIQXA6OC-OBnUwAVhh5LZ-kCwRpdIZkWicVyiwA9AAqOCW836cnkoA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge1tiacTJTIAhtEK0yHJgBNK%2BSpPRRE0aB2iRwYAL4gtQA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false

### Repro Code

```TypeScript
export class Game {
  private isRunning: boolean;

  constructor() {
    this.isRunning = false;
  }

  public start(): void {
    if (this.isRunning) {
      throw new Error('Could not start Game: Already running!');
    }
    this.isRunning = true;

    let time = 0;

    while (this.isRunning) {
      while (time >= 0) {
        this.update();
        time++;

        if (!this.isRunning) {
          break;
        }
      }
      this.render();
    }
  }

  public update(): void {
      this.isRunning = Math.random() > 0.5;
  }

  public render(): void {
    /* */
  }
}
```

### ESLint Config

```javascript
module.exports = {
  "rules": {
    "@typescript-eslint/no-unnecessary-condition": "error"
  }
}
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    "strictNullChecks": true
  }
}
```

### Expected Result

It should report no error.

### Actual Result

Reports errors on line 16 and 21 (see Playground link).

### Additional Info

I would expect the [no-unnecessary-condition] rule is disabled when dealing with class instance properties as they can be changed anytime.

**Comment by bradzacher** [comment]
https://typescript-eslint.io/rules/no-unnecessary-condition/#values-modified-within-function-calls

Check the types that TS reports for the property. You'll notice that it's reporting `this.isRunning: true` - hence the rule is reporting. See the above FAQ on the rule's docs.
